gpytorch/lazy/added_diag_lazy_tensor.py

```diff
@@ -125,3 +129,21 @@ def _init_cache_for_non_constant_diag(self, eye, batch_shape, n):
         logdet = self._r_cache.diagonal(dim1=-1, dim2=-2).abs().log().sum(-1).mul(2)
         logdet -= (1.0 / self._noise).log().sum([-1, -2])
         self._precond_logdet_cache = logdet.view(*batch_shape) if len(batch_shape) else logdet.squeeze()
+
+    @cached(name="svd")
+    def _svd(self) -> Tuple["LazyTensor", Tensor, "LazyTensor"]:
+        diag = self._diag_tensor.diag()
+        if torch.equal(diag, diag[..., :1].expand(diag.shape)):
+            U, S_, V = self._lazy_tensor.svd()
+            S = S_ + diag  # this assumes all diagonal entries are positive
+            return U, S, V
```

Comment:
We probably don't want to include these because the backwards call of the SVD of the matrix plus the diagonal doesn't necessarily match the result of this operation, due to gradient un-identifiability. This should be what's causing the test failures.

In this case, it's also blindly assuming that the matrix is PSD, which may or may not be true; shifting doesn't necessarily hold for SVD of non PD matrices (thanks Jake for pointing that one out).

```suggestion

```

Reply:
I mean isn't the standing assumption for lazies throughout that they are psd? At least prior to my `TriangularLazyTensor` PR...

Reply:
My main issue is that then the most efficient implementation of SVD is to just call symeig.

Reply:
I think so, but maybe @jacobrgardner would know of some cases where they are not necessarily?

Reply:
Fair enough - but aren't there still some serious issues with `symeig` speed on smaller matrices, especially on the GPU?